crates/hotshot/src/traits/election/randomized_committee_members.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2021-2024 Espresso Systems (espressosys.com)
+// This file is part of the HotShot repository.
+
+// You should have received a copy of the MIT License
+// along with the HotShot repository. If not, see <https://mit-license.org/>.
+
+use std::{
+    cmp::max,
+    collections::{BTreeMap, BTreeSet},
+    marker::PhantomData,
+    num::NonZeroU64,
+};
+
+use ethereum_types::U256;
+use hotshot_types::{
+    traits::{
+        election::Membership,
+        network::Topic,
+        node_implementation::{ConsensusTime, NodeType},
+        signature_key::{SignatureKey, StakeTableEntryType},
+    },
+    PeerConfig,
+};
+use rand::{rngs::StdRng, Rng};
+use utils::anytrace::Result;
+
+use crate::traits::election::helpers::QuorumFilterConfig;
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+/// The static committee election
+pub struct RandomizedCommitteeMembers<T: NodeType, C: QuorumFilterConfig> {
+    /// The nodes eligible for leadership.
+    /// NOTE: This is currently a hack because the DA leader needs to be the quorum
+    /// leader but without voting rights.
+    eligible_leaders: Vec<<T::SignatureKey as SignatureKey>::StakeTableEntry>,
+
+    /// The nodes on the committee and their stake
+    stake_table: Vec<<T::SignatureKey as SignatureKey>::StakeTableEntry>,
+
+    /// The nodes on the committee and their stake, indexed by public key
+    indexed_stake_table:
+        BTreeMap<T::SignatureKey, <T::SignatureKey as SignatureKey>::StakeTableEntry>,
+
+    /// The network topic of the committee
+    committee_topic: Topic,
+
+    /// Phantom
+    _pd: PhantomData<C>,
+}
+
+impl<TYPES: NodeType, CONFIG: QuorumFilterConfig> RandomizedCommitteeMembers<TYPES, CONFIG> {
+    /// Creates a set of indices into the stake_table which reference the nodes selected for this epoch's committee
+    fn make_quorum_filter(&self, epoch: <TYPES as NodeType>::Epoch) -> BTreeSet<usize> {
+        CONFIG::execute(epoch.u64(), self.stake_table.len())
+    }
+}
+
+impl<TYPES: NodeType, CONFIG: QuorumFilterConfig> Membership<TYPES>
+    for RandomizedCommitteeMembers<TYPES, CONFIG>
+{
+    type Error = utils::anytrace::Error;
+
+    /// Create a new election
+    fn new(
+        eligible_leaders: Vec<PeerConfig<<TYPES as NodeType>::SignatureKey>>,
+        committee_members: Vec<PeerConfig<<TYPES as NodeType>::SignatureKey>>,
+        committee_topic: Topic,
+    ) -> Self {
+        // The only time these two are unequal is in a benchmarking scenario that isn't covered by this test case
+        assert_eq!(
+            eligible_leaders, committee_members,
+            "eligible_leaders should be the same as committee_members!"
+        );
```

Comment:
Unless the committees have been unified (across DA/quorum) yet, we expect eligible leaders to be different than the full list of members in the DA committee. This is because we want to enforce that quorum leaders == da leaders for now

Reply:
I think this is a bit weird because `committee_members` here doesn't mean DA (the word `committee` is probably the wrong word here). the DA committee is currently the `committee_members` of `da_membership`, but @tbro is unifying the 2 by eliminating the `Memberships` wrapper struct

I think the only thing `eligible_leaders` is used for is VID benchmarking with a fixed leader